### ISSUE TITLE
@export_category("") name Strings cause conflicts with like Class names and Autoloads

### ISSUE THREAD

**Issue by Dabnabbit** [godotengine/godot#80634]
### Godot version

4.1.1 Stable

### System information

Windows 11, RTX 3070TI, Vulcan

### Issue description

This may or may not be a bug, but definitely needs some kind of detection or warning build into the code if it is intentional or unavoidable.

After bashing my head against a wall today trying to figure out why a single specific script would not utilize referenced autoload function/methods and trying to figure out why it was the only script that would not recognize the autoload class, I realized that I had an organizational @export_category() at the top using the same name. Apparently this means that any future references to that same string will not properly reference the class or autoload.

Example:

Having an autoload script named "Debug" with a method "draw_cube()" in it.
Then in another script having a parameter organizer like such:
```
@export_category("Debug")
@export_range(0.0, 1.0) var debug_polyline_pos : float = 0.0
```

and then later attempting
```
Debug.draw_cube(Vector3.ZERO, Vector3.ZERO, 2.0, Color.PURPLE)
```
in that same script will result the inability for the game to recognize Debug as the autoload script and will not find the .draw_cube() method.

This may be expected/intentional behavior, but if that is the case this definitely needs to be outlined and/or flagged in the script-editor as a conflict.

Also as a side note, I love the fact the export_category("Debug") specifically has a little built-in easter-egg for using the term "Debug" so if there's anyway we can keep that around or maybe even expose the ability for users to control that little icon that would be pretty amazing!

**Comment by AThousandShips** [comment]
Thank you for reporting, this has been reported previously and resolved in #78254, which has not been cherry picked yet

**Comment by AThousandShips** [comment]
Duplicate of:
* https://github.com/godotengine/godot/issues/67865
* https://github.com/godotengine/godot/issues/73843
